src/controller/python/chip/clusters/ClusterObjects.py

```diff
@@ -92,26 +143,41 @@ def TagDictToLabelDict(self, debugPath: str, tlvData: Dict[int, Any]) -> Dict[st
                 # We do not have enough infomation for this field.
                 ret[tag] = value
                 continue
-            if descriptor.IsArray:
+
+            if (value == None):
+                ret[descriptor.Label] = NullValue
+                continue
+
+            if (typing.get_origin(descriptor.Type) == typing.Union):
+                realType = GetUnionUnderlyingType(descriptor.Type)
+                if (realType == None):
+                    raise ValueError(
+                        f"Field {debugPath}.{self.Label} has no real type")
```

Comment:
`real type` is a bit misleading since a `real type` may refer to "real number type". Consider using `underlying type` directly?

BTW, if we reached here, seems the codegen is emitting something wrong. Can we add a check in the constructor of the ClusterObject class?